### ISSUE TITLE
Unable to create more than one consumer in the Consumers dashboard page

### ISSUE THREAD

**Issue by jamelseagraves** [pantsel/konga#112]
I am running Docker containers with Konga 0.8.7 and Kong 0.10.3. In the Consumers page of Konga, I was able to use the "**+ Create Consumer**" button once to create a new consumer. After that, I cannot create another consumer. I get the following output with a 500 response error in the Konga container logs:

```
10/6/2017 7:02:39 AM  toJSON: [Function: responseToJSON],
10/6/2017 7:02:39 AM  caseless:
10/6/2017 7:02:39 AM   Caseless {
10/6/2017 7:02:39 AM     dict:
10/6/2017 7:02:39 AM      { date: 'Fri, 06 Oct 2017 14:02:39 GMT',
10/6/2017 7:02:39 AM        'content-type': 'application/json; charset=utf-8',
10/6/2017 7:02:39 AM        'transfer-encoding': 'chunked',
10/6/2017 7:02:39 AM        connection: 'close',
10/6/2017 7:02:39 AM        'access-control-allow-origin': '*',
10/6/2017 7:02:39 AM        server: 'kong/0.10.3' } },
10/6/2017 7:02:39 AM  read: [Function],
10/6/2017 7:02:39 AM  body: { message: 'An unexpected error occurred' },
10/6/2017 7:02:39 AM  code: 500,
10/6/2017 7:02:39 AM  status: 500,
10/6/2017 7:02:39 AM  statusType: 5,
10/6/2017 7:02:39 AM  info: false,
10/6/2017 7:02:39 AM  ok: false,
10/6/2017 7:02:39 AM  clientError: false,
10/6/2017 7:02:39 AM  serverError: true,
10/6/2017 7:02:39 AM  error:
10/6/2017 7:02:39 AM   { Error: got 500 response
10/6/2017 7:02:39 AM       at generateErrorMessage (/app/node_modules/unirest/index.js:399:23)
10/6/2017 7:02:39 AM       at Request.handleRequestResponse [as _callback] (/app/node_modules/unirest/index.js:402:13)
10/6/2017 7:02:39 AM       at Request.self.callback (/app/node_modules/unirest/node_modules/request/request.js:187:22)
10/6/2017 7:02:39 AM       at emitTwo (events.js:106:13)
10/6/2017 7:02:39 AM       at Request.emit (events.js:191:7)
10/6/2017 7:02:39 AM       at Request.<anonymous> (/app/node_modules/unirest/node_modules/request/request.js:1044:10)
10/6/2017 7:02:39 AM       at emitOne (events.js:96:13)
10/6/2017 7:02:39 AM       at Request.emit (events.js:188:7)
10/6/2017 7:02:39 AM       at IncomingMessage.<anonymous> (/app/node_modules/unirest/node_modules/request/request.js:965:12)
10/6/2017 7:02:39 AM       at emitNone (events.js:91:20)
10/6/2017 7:02:39 AM       at IncomingMessage.emit (events.js:185:7)
10/6/2017 7:02:39 AM       at endReadableNT (_stream_readable.js:974:12)
10/6/2017 7:02:39 AM       at _combinedTickCallback (internal/process/next_tick.js:80:11)
10/6/2017 7:02:39 AM       at process._tickDomainCallback (internal/process/next_tick.js:128:9) status: 500 },
10/6/2017 7:02:39 AM  created: false,
10/6/2017 7:02:39 AM  accepted: false,
10/6/2017 7:02:39 AM  nonAuthoritativeInformation: false,
10/6/2017 7:02:39 AM  noContent: false,
10/6/2017 7:02:39 AM  resetContent: false,
10/6/2017 7:02:39 AM  partialContent: false,
10/6/2017 7:02:39 AM  multiStatus: false,
10/6/2017 7:02:39 AM  alreadyReported: false,
10/6/2017 7:02:39 AM  imUsed: false,
10/6/2017 7:02:39 AM  multipleChoices: false,
10/6/2017 7:02:39 AM  movedPermanently: false,
10/6/2017 7:02:39 AM  found: false,
10/6/2017 7:02:39 AM  seeOther: false,
10/6/2017 7:02:39 AM  notModified: false,
10/6/2017 7:02:39 AM  useProxy: false,
10/6/2017 7:02:39 AM  temporaryRedirect: false,
10/6/2017 7:02:39 AM  permanentRedirect: false,
10/6/2017 7:02:39 AM  badRequest: false,
10/6/2017 7:02:39 AM  unauthorized: false,
10/6/2017 7:02:39 AM  paymentRequired: false,
10/6/2017 7:02:39 AM  forbidden: false,
10/6/2017 7:02:39 AM  notFound: false,
10/6/2017 7:02:39 AM  methodNotAllowed: false,
10/6/2017 7:02:39 AM  notAcceptable: false,
10/6/2017 7:02:39 AM  proxyAuthenticationRequired: false,
10/6/2017 7:02:39 AM  requestTimeout: false,
10/6/2017 7:02:39 AM  conflict: false,
10/6/2017 7:02:39 AM  gone: false,
10/6/2017 7:02:39 AM  lengthRequired: false,
10/6/2017 7:02:39 AM  preconditionFailed: false,
10/6/2017 7:02:39 AM  requestEntityTooLarge: false,
10/6/2017 7:02:39 AM  uriTooLong: false,
10/6/2017 7:02:39 AM  unsupportedMediaType: false,
10/6/2017 7:02:39 AM  rangeNotSatisfiable: false,
10/6/2017 7:02:39 AM  expectationFailed: false,
10/6/2017 7:02:39 AM  misdirectedRequest: false,
10/6/2017 7:02:39 AM  unprocessableEntity: false,
10/6/2017 7:02:39 AM  locked: false,
10/6/2017 7:02:39 AM  failedDependency: false,
10/6/2017 7:02:39 AM  upgradeRequired: false,
10/6/2017 7:02:39 AM  preconditionRequired: false,
10/6/2017 7:02:39 AM  tooManyRequests: false,
10/6/2017 7:02:39 AM  requestHeaderFieldsTooLarge: false,
10/6/2017 7:02:39 AM  internalServerError: true,
10/6/2017 7:02:39 AM  notImplemented: false,
10/6/2017 7:02:39 AM  badGateway: false,
10/6/2017 7:02:39 AM  serviceUnavailable: false,
10/6/2017 7:02:39 AM  gatewayTimeout: false,
10/6/2017 7:02:39 AM  httpVersionNotSupported: false,
10/6/2017 7:02:39 AM  variantAlsoNegotiates: false,
10/6/2017 7:02:39 AM  insufficientStorage: false,
10/6/2017 7:02:39 AM  loopDetected: false,
10/6/2017 7:02:39 AM  notExtended: false,
10/6/2017 7:02:39 AM  cookies: {},
10/6/2017 7:02:39 AM  cookie: [Function],
10/6/2017 7:02:39 AM  raw_body: { message: 'An unexpected error occurred' } }
```

I also checked the error in the browser, and this is the message output:

```
Failed to create consumer		angular.js:14199

config:
	cached: false
	data:
		custom_id: ""
		token: "eyJhbGciOiJIUzI1NiJ9.MQ.Lu-KcR4aCeuT9hi1K474zV3s4VaopLDCcf4nZvH6DQo"
		username: "administrator"
		__proto__: Object
	headers:
		Accept: "application/json, text/plain, */*"
		Content-Type: "application/json;charset=utf-8"
		authorization: "Bearer eyJhbGciOiJIUzI1NiJ9.MQ.Lu-KcR4aCeuT9hi1K474zV3s4VaopLDCcf4nZvH6DQo"
		__proto__: Object
		method: "POST"
		transformRequest: [ƒ]
		transformResponse: [ƒ]
		url: "/kong/consumers"
		__proto__: Object
	data:
		body: {message: "An unexpected error occurred"}
		headers:
			access-control-allow-origin: "*"
			connection: "close"
			content-type: "application/json; charset=utf-8"
			date: "Fri, 06 Oct 2017 14:02:39 GMT"
			server: "kong/0.10.3"
			transfer-encoding: "chunked"
			__proto__: Object
		request:
			headers:
				Content-Type: "application/json"
				accept: "application/json"
				content-length: 43
				__proto__: Object
				method: "post"
				uri:
					auth: null
					hash: null
					host: "[my hostname]:8001"
					hostname: "[my hostname]"
					href: "http://[my ip address]:8001/consumers"
					path: "/consumers"
					pathname: "/consumers"
					port: "8001"
					protocol: "http:"
					query: null
					search: null
					slashes: true
					__proto__: Object
				__proto__: Object
			statusCode: 500
			__proto__: Object
		headers: ƒ (name)
		status: 500
		statusText: undefined
		__proto__: Object
```

Any help on this would be great. Thanks!

**Comment by pantsel** [comment]
Hello @jamelseagraves, have you tried to reproduce that behaviour with the 0.11 version of kong?

I can't seem to be able to reproduce it. Are you logged in Konga with an administrator account?

If not, what are the configured settings?

**Comment by jamelseagraves** [comment]
@pantsel thanks for the reply! Yes, I just tried to reproduce it with kong version 0.11.0, and the issue is still the same. I am logged in as the default admin user. As before, I can create 1 consumer, but then it fails with the response 500 error when I try to create a second consumer.

Also FYI, when I first started konga and added a new connection to my kong server, the kong version for the connection in the Connections page of the dashboard showed up as `0.10.x`. I am not sure if you are actually detecting which kong version is running or if you are just setting it to a default value, but I figured I would share this as well.

**Comment by pantsel** [comment]
@jamelseagraves , i've fixed some issues in the upcoming version of Konga.

Could you try using the [next branch](https://github.com/pantsel/konga/tree/next) or <code>$docker pull pantsel/konga:next</code> and see if that issue still persists?

Thanks in advance.

**Comment by jamelseagraves** [comment]
@pantsel I just tried using the `pantsel/konga:next`, and the issue persists. I am using postgres as the db, following your tutorial example. Are you able to create multiple consumers if you integrate postgres with konga?

**Comment by pantsel** [comment]
@jamelseagraves ,

you should be able to create multiple consumers because they have nothing to do with Konga's database.

Either way I will look into it and let you know.

Thanks for the feedback!

**Comment by pantsel** [comment]
@jamelseagraves, i think i've identified the issue.

Apparently Kong's consumers api can't handle sending empty properties for username and client_id althought you can do it one time...

Pretty crazy, huh? :) 
It also doesn't provide a proper response.

If that's the case and I'm not mistaken, I've pushed an update at the `next` branch.

Would you be kind enough to test it and let me know?

**Comment by jamelseagraves** [comment]
@pantsel thanks for jumping right on this! I believe Kong's consumer api is supposed to require that either username or client_id be populated, at least according to the docs. Very odd.

I just tested your new `next` image, and it seems to be working. I will wait for an official new docker image version release. Thanks again!

**Comment by pantsel** [comment]
Fixed on [v0.8.8](https://github.com/pantsel/konga/releases/tag/0.8.8)
